File: matlab/noiseParamInit.m
function noise = noiseParamInit(noise, y)

% NOISEPARAMINIT Noise model's parameter initialisation.

% NOISE

% If this flag is set then the noise model leads to constant values of
% beta (e.g. a Gaussian with constant variance for each data-point &
% dimension). The default setting is set here as false.
noise.spherical = 0;
noise.logconcave = 1;

fhandle = str2func([noise.type 'NoiseParamInit']);
if nargin > 1
  noise = fhandle(noise, y);
else
  noise = fhandle(noise);
end

File: R/R/noiseParamInit.R
## Not implemented: mgaussianNoiseParamInit, ncnmNoiseParamInit,
## ngaussNoiseParamInit, noiseParamInit, orderedNoiseParamInit,
## probitNoiseParamInit, scaleNoiseParamInit
noiseParamInit <- function(noise, y) {
  noise$spherical = 0
  noise$logconcave = 1

  fhandle <- get(paste(noise$type, "NoiseParamInit", sep=""), mode="function")
  if (nargs() > 1)
    noise = fhandle(noise, y)
  else
    noise = fhandle(noise)

  return (noise)
}
